File: backend/entity/member.go
package entity

import "gorm.io/gorm"

type Member struct {
	gorm.Model
	Username  string    `gorm:"uniqueIndex" json:"username"`
	Password  string    `json:"password"`
	Email     string    `gorm:"uniqueIndex" json:"email"`
	Histories []History `gorm:"foreignKey:MemberID" json:"histories"`
}

File: frontend/src/interfaces/Member.ts
import type { History } from "./History";

export interface Member {
  ID: number;
  username: string;
  password?: string;
  email: string;
  histories?: History[];
}

export interface CreateMemberRequest {
  username: string;
  password: string;
  email: string;
}

export interface LoginMemberRequest {
  username: string;
  password: string;
}

// Keep old interfaces for backward compatibility
export interface MemberInterface {
  ID?: number;
  username: string;
  password: string;
  email: string;
  histories?: HistoryInterface[];
}

export interface HistoryInterface {
  ID?: number;
  playedAt?: Date;
  soundID?: number;
  sound?: string;
  memberID?: number;
  member?: string;
}
